enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).my_each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_all?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_any?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
```

Comment:
my_none should behave the same way the original method does:

- when no block or argument is given returns true only if none of the collection members is true.
`[1,2,3,4].my_none?` should return false. Right it gives a no block given error.

- when a class is passed as an argument returns true if none of the collection is a member of such class
`[1,2,3].none? (String)`
`=> true`
`[1,2,3].none? (Integer)`
`=> false`

- when a Regex is passed as an argument returns true only if none of the collection matches the Regex
['ant', 'bear', 'cat'].none?(/d/)                        #=> true

- when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`[nil, true, 99].any?                              #=> true`